### PR TITLE
#3715: [gatsby-plugin-catch-links] Do not catch same domain links in a pathPrefix case

### PR DESCRIPTION
**Pull request by nadiia**
Do not catch same domain links if there is a pathPrefix used in the app but the pathPrefix not included in a same domain link.

### CODE REVIEW

File: packages/gatsby-plugin-catch-links/src/catch-links.js [hunk 1]
```diff
@@ -56,6 +58,9 @@ module.exports = function(root, cb) {
 
     if (a1.host !== a2.host) return true
 
+    if (!`${a1.host}${a1.pathname}`.startsWith(`${a2.host}${withPrefix(`/`)}`))
```
Comment (KyleAMathews) [inline packages/gatsby-plugin-catch-links/src/catch-links.js]: `startsWith` isn't supported on IE https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

**Comment by nadiia** [comment]
@KyleAMathews sure. It may not be a direct solution for #3715 (it seems like they should not use absolute links to internal links). 

It handles the case when a `pathPrefix` used in an app, and there happen to be a link referring to the same domain but outside of the app's `pathPrefix`. 

For example, we have an app that lives at `https://example.com/myapp/`, with the `pathPrefix` set to `myapp` respectively. 
When adding any link (absolute link) to the same domain but outside of the `/myapp` path, for example, `<a href="https://example.com/not-my-app">` the plugin catches it and treats it as a relative link navigating to `https://example.com/myapp/https://example.com/not-my-app`.

An additional condition in this PR handles the case described and will navigate to an external link instead of doing a `pushState`.

**Comment by KyleAMathews** [comment]
Ah gotcha! Ok, yeah this totally makes sense.

Could you add a comment explaining the use case? Plus please replace `startsWith` with a `slice` based approach (or whatever seems appropriate).

**Comment by nadiia** [comment]
updated, please take a look @KyleAMathews

**Comment by KyleAMathews** [comment]
Thanks! Looks great

**Pull request merged**